### PR TITLE
Added Jest and Enzyme for client side Testing

### PR DESCRIPTION
**Pull request by js-sapphire**
#8 @tanaypratap 
Hi, I have tried adding all that was mentioned in the issue. Could you please take a look at the PR. :)

### CODE REVIEW

File: client/__tests__/subapp.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import {SubApp} from '../app/subapp';
+import { shallow } from 'enzyme';
+
+test("renders the heading with props supplied", () => {
+    const result = shallow(<SubApp message="Tanay"/>).contains(<h1>Hello Tanay</h1>);
```
[the remark this answers is not in the document]
Comment (js-sapphire) [reply client/__tests__/subapp.spec.tsx]: On it

File: client/__tests__/subapp.spec.tsx [hunk 2]
```diff
@@ -3,11 +3,11 @@ import {SubApp} from '../app/subapp';
 import { shallow } from 'enzyme';
 
 test("renders the heading with props supplied", () => {
-    const result = shallow(<SubApp message="Tanay"/>).contains(<h1>Hello Tanay</h1>);
+    const result = shallow(<SubApp message="Polaris"/>).contains(<h1>Hello Tanay</h1>);
```
Comment (tanaypratap) [inline client/__tests__/subapp.spec.tsx]: Wouldn't this test fail now? when it tries to match Polaris with Tanay?

File: client/__tests__/subapp.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import {SubApp} from '../app/subapp';
```
Comment (tanaypratap) [inline client/__tests__/subapp.spec.tsx]: consistency in  import is not there. Give spaces after bracket. { SubApp }

File: yarn-error.log [hunk 4]
```diff
@@ -0,0 +1,6202 @@
+Arguments:
```
Comment (tanaypratap) [inline yarn-error.log]: this file doesn't need to be committed. Delete this. Also, can you add this to .gitignore?

File: tsconfig.json [hunk 5]
```diff
@@ -4,6 +4,7 @@
     "target": "es6",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017','ES2018' or 'ESNEXT'. */
     "module": "commonjs",                     /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */
     // "allowJs": true,                       /* Allow javascript files to be compiled. */
+    "jsx": "react",
     // "checkJs": true,                       /* Report errors in .js files. */
     // "jsx": "preserve",                     /* Specify JSX code generation: 'preserve', 'react-native', or 'react'. */
```
Comment (tanaypratap) [inline tsconfig.json]: delete this entry if you're using "react"

File: package.json [hunk 6]
```diff
@@ -11,7 +11,8 @@
     "start": "webpack-dev-server -d --content-base ./client/public --config ./client/webpack.config.js",
     "start:server": "npm run build:watch",
     "build:server": "tsc",
-    "build:watch": "nodemon --exec ./node_modules/.bin/ts-node -- ./_server/index.ts"
+    "build:watch": "nodemon --exec ./node_modules/.bin/ts-node -- ./_server/index.ts",
+    "test": "jest"
```
Comment (tanaypratap) [inline package.json]: jest has command for: watch mode, for updating snapshots. Please check and create script entry for those as well.

File: tsconfig.json (hunk 5 above)
Comment (js-sapphire) [reply tsconfig.json]: [Issue](https://github.com/Microsoft/vscode/issues/15814)

**Pull request merged**